lib/purescript-cst/src/Language/PureScript/CST/Parser.y

```diff
@@ -304,16 +297,12 @@ type2 :: { Type () }
   | type3 '=>' type1 {% do cs <- toConstraint $1; pure $ TypeConstrained () cs $2 $3 }
 
 type3 :: { Type () }
-  : type4 { $1 }
+  : type4 %shift { $1 }
   | type3 qualOp type4 { TypeOp () $1 (getQualifiedOpName $2) $3 }
```

Comment:
This'll fix your last conflict:

```suggestion
  | type3 qualOp type4 %shift { TypeOp () $1 (getQualifiedOpName $2) $3 }
```

Reply:
Thanks!
How were you able to figure that out?

Reply:
In this case, the `type4 : type5 %shift` production told Happy to disregard this reduction if anything under `type5` can keep being greedy. Removing that production meant that productions that end in `type4` would themselves need to be given `%shift` precedence, or else the shift-reduce conflicts being prevented by the `%shift` in `type4 : type5 %shift` would re-emerge (which is, I assume, what happened here).

Reply:
Ah! I see! That makes sense. Thanks for explaining that. :smiley: